addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -172,18 +176,47 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             case CHANNEL_SETTEMP:
                 if (command instanceof DecimalType) {
                     BigDecimal temperature = new BigDecimal(command.toString());
-                    if (temperature.compareTo(HeatingModel.TEMP_MIN) == -1) {
-                        temperature = HeatingModel.TEMP_MIN;
-                    } else if (temperature.compareTo(HeatingModel.TEMP_MAX) == 1) {
-                        temperature = HeatingModel.TEMP_MAX;
-                    }
-                    fritzBox.setSetTemp(ain, temperature.divide(HeatingModel.TEMP_FACTOR));
+                    fritzBox.setSetTemp(ain, HeatingModel.fromCelsius(temperature));
                 } else if (command instanceof OnOffType) {
-                    BigDecimal temperature = command.equals(OnOffType.ON) ? HeatingModel.TEMP_ON
-                            : HeatingModel.TEMP_OFF;
+                    BigDecimal temperature = command.equals(OnOffType.ON) ? HeatingModel.TEMP_FRITZ_ON
+                            : HeatingModel.TEMP_FRITZ_OFF;
                     fritzBox.setSetTemp(ain, temperature);
                 }
                 break;
+            case CHANNEL_RADIATOR_MODE:
+                if (command instanceof StringType) {
+                    String mode = command.toString();
+                    if (command.equals(MODE_ON)) {
+                        BigDecimal settemp = (BigDecimal) getThing().getConfiguration().get(THING_SETTEMP);
```

Comment:
I see a lot of repetition (one or two patterns)

1) (optionally) fetch configuration
2) call setSetTemp with a fixed value or the configuration value above (if needed converted from celcius)
3) call updateState with a fixed value or the configuration value above (if needed converted to celcius)

Btw:

Should/can this:

```java
Channel channelSetTemp = getThing().getChannel(CHANNEL_SETTEMP);
updateState(channelSetTemp.getUID(), new DecimalType(settemp));
```

be written as:

```java
updateState(CHANNEL_SETTEMP, new DecimalType(settemp));
```